From d8a3b256da66e2e625691c39e67e6fa039fdf0d7 Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Mon, 13 Apr 2015 14:44:35 -0700
Subject: [PATCH 05/14] Media: Build VaapiPictureWayland as part of Media.

This is temporary, so we can avoid gl circular dependency. Eventually,
we want to be as aligned as possible with upstream Media solution and need
to figure out how we could use upstream VaapiWrapper (while able to generate
custom va_stubs if possible?).
---
 content/content_common.gypi | 51 ++++++++++++++++++++++++++++-----------------
 1 file changed, 32 insertions(+), 19 deletions(-)

diff --git a/content/content_common.gypi b/content/content_common.gypi
index 5593097..31eeb74 100644
--- a/content/content_common.gypi
+++ b/content/content_common.gypi
@@ -826,7 +826,7 @@
         },
       ],
     }],
-    ['chromeos==1', {
+    ['chromeos==1 or use_ozone==1', {
       'sources': [
         'common/gpu/media/accelerated_video_decoder.h',
         'common/gpu/media/h264_decoder.cc',
@@ -875,7 +875,7 @@
         'common/gpu/media/tegra_v4l2_device.h',
       ],
     }],
-    ['target_arch != "arm" and chromeos == 1', {
+    ['target_arch != "arm" and use_ozone == 1', {
       'dependencies': [
         '../media/media.gyp:media',
         '../third_party/libyuv/libyuv.gyp:libyuv',
@@ -884,18 +884,20 @@
         'common/gpu/media/va_surface.h',
         'common/gpu/media/vaapi_jpeg_decoder.cc',
         'common/gpu/media/vaapi_jpeg_decoder.h',
-        'common/gpu/media/vaapi_picture.cc',
-        'common/gpu/media/vaapi_picture.h',
         'common/gpu/media/vaapi_video_decode_accelerator.cc',
         'common/gpu/media/vaapi_video_decode_accelerator.h',
-        'common/gpu/media/vaapi_video_encode_accelerator.cc',
-        'common/gpu/media/vaapi_video_encode_accelerator.h',
-        'common/gpu/media/vaapi_wrapper.cc',
-        'common/gpu/media/vaapi_wrapper.h',
+        '<(DEPTH)/ozone/media/vaapi_picture.cc',
+        '<(DEPTH)/ozone/media/vaapi_picture.h',
+        '<(DEPTH)/ozone/media/vaapi_picture_wayland.cc',
+        '<(DEPTH)/ozone/media/vaapi_picture_wayland.h',
+        '<(DEPTH)/ozone/media/vaapi_wrapper.cc',
+        '<(DEPTH)/ozone/media/vaapi_wrapper.h',
+
       ],
       'conditions': [
         ['use_x11 == 1', {
           'variables': {
+            'extra_header': 'common/gpu/media/va_stub_header.fragment',
             'sig_files': [
               'common/gpu/media/va.sigs',
               'common/gpu/media/va_x11.sigs',
@@ -906,21 +908,32 @@
             'common/gpu/media/vaapi_tfp_picture.h',
           ],
         }, {
-          'variables': {
-            'sig_files': [
-              'common/gpu/media/va.sigs',
-              'common/gpu/media/va_drm.sigs',
-            ],
-          },
-          'sources': [
-            'common/gpu/media/vaapi_drm_picture.cc',
-            'common/gpu/media/vaapi_drm_picture.h',
-          ],
+          'conditions': [
+            ['ozone_platform_wayland == 1', {
+              'variables': {
+                'extra_header': '../ozone/media/va_wayland_stub_header.fragment',
+                'sig_files': [
+                  '../ozone/media/va_wayland.sigs',
+                ],
+              },
+            }, {
+              'variables': {
+                'extra_header': 'common/gpu/media/va_stub_header.fragment',
+                'sig_files': [
+                  'common/gpu/media/va.sigs',
+                  'common/gpu/media/va_drm.sigs',
+                ],
+              },
+              'sources': [
+                'common/gpu/media/vaapi_drm_picture.cc',
+                'common/gpu/media/vaapi_drm_picture.h',
+              ],
+            }],
+          ]
         }],
       ],
       'variables': {
         'generate_stubs_script': '../tools/generate_stubs/generate_stubs.py',
-        'extra_header': 'common/gpu/media/va_stub_header.fragment',
         'outfile_type': 'posix_stubs',
         'stubs_filename_root': 'va_stubs',
         'project_path': 'content/common/gpu/media',
-- 
1.9.1

